rclpy/src/rclpy/_rclpy.c

```diff
@@ -1193,20 +1193,33 @@ rclpy_create_timer(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
+  rcl_clock_t * clock = (rcl_clock_t *) PyMem_Malloc(sizeof(rcl_clock_t));
```

Comment:
I see that `timer` already gets allocated with `PyMem_Malloc`, but what I'm wondering is why we do this rather use the `rcl_allocator_t`? Maybe @sloretz or whoever did this originally has an idea. Even if we don't have to use `PyMem_Malloc`, it might be the right preference anyways. It's just that we're mixing the two, which happen to be the same right now, but are not necessarily both using `malloc` under the hood.

We control both the allocation and free of these resources, so either should work I think. We're never relying on Python to free them for us (I think).

Reply:
I don't have a good reason for using this or `mallloc` via an `rcl_allocator_t`.  There is more info about [pymalloc here](https://docs.python.org/3/c-api/memory.html#pymalloc). As long as memory is free'd by the same allocator it was allocated with then I wouldn't expect problems from mixing the two.

> We control both the allocation and free of these resources, so either should work I think. We're never relying on Python to free them for us (I think).

AFAIK this is correct.